packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -323,30 +323,43 @@ abstract class FlutterCommand extends Command<void> {
   }
 
   BuildMode getBuildMode() {
-    final List<bool> modeFlags = <bool>[argResults['debug'], argResults['profile'], argResults['release']];
-    if (modeFlags.where((bool flag) => flag).length > 1)
+    bool debug;
```

Comment:
I've updated getBuildMode to not throw if used in a command that does not add the debug, profile, or release arguments.